sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/higherOrderFunctions.scala

```diff
@@ -74,7 +76,7 @@ case class ResolveLambdaVariables(conf: SQLConf) extends Rule[LogicalPlan] {
   private val canonicalizer = {
     if (!conf.caseSensitiveAnalysis) {
       // scalastyle:off caselocale
-      s: String => s.toLowerCase
+      s: String => s.toLowerCase(Locale.ROOT)
```

Comment:
Not sure about this one. It's affecting variable names in a user's SQL? then it shouldn't change